geoh5py/objects/surveys/direct_current.py

```diff
@@ -229,36 +231,18 @@ def current_electrodes(self):
     def default_type_uid(cls) -> uuid.UUID:
         """Default unique identifier. Implemented on the child class."""
 
-    @property
-    def metadata(self):
-        """
-        Metadata attached to the entity.
-        """
-        if getattr(self, "_metadata", None) is None:
-            metadata = self.workspace.fetch_metadata(self.uid)
-            self._metadata = metadata
-        return self._metadata
-
-    @metadata.setter
-    def metadata(self, values):
-        if not len(values) == 2:
-            raise ValueError(
-                f"Metadata must have two key-value pairs. {values} provided."
-            )
-
-        default_keys = ["Current Electrodes", "Potential Electrodes"]
-
-        if list(values.keys()) != default_keys:
-            raise ValueError(f"Input metadata must have for keys {default_keys}")
-
-        if self.workspace.get_entity(values["Current Electrodes"])[0] is None:
-            raise IndexError("Input Current Electrodes uuid not present in Workspace")
+    @Curve.metadata.setter  # type: ignore
+    def metadata(self, values: dict | None):
+        if isinstance(values, dict):
+            default_keys = ["Current Electrodes", "Potential Electrodes"]
+            if sorted(list(values.keys())) != default_keys:
+                raise ValueError(f"Input metadata must have for keys {default_keys}")
 
-        if self.workspace.get_entity(values["Potential Electrodes"])[0] is None:
-            raise IndexError("Input Potential Electrodes uuid not present in Workspace")
+            for key in default_keys:
+                if self.workspace.get_entity(values[key])[0] is None:
+                    raise IndexError(f"Input {key} uuid not present in Workspace")
 
-        self._metadata = values
-        self.workspace.update_attribute(self, "metadata")
+        super(Curve, Curve).metadata.fset(self, values)  # type: ignore
```

Comment:
Not sure this will work generally with more metadata then just the `default_keys`. Can you augment the tests under `tests/survey_dcip_test.py` to check?

Reply:
Now, metadata must have those two parameters for direct_current. 
Tell me what you think about the implementation.
It's tested

Reply:
Yep, logic looks good.